scripts/images/upload_image.py

```diff
@@ -1,91 +1,76 @@
 
 
-def main(argv):
-    global cos
-    apikey = ''
-    instanceid = ''
-    region = ''
-    bucketname = ''
-    infile = ''
-    objectname = ''
-    try:
-        (opts, args) = getopt.getopt(argv, 'hk:i:r:b:f:o:', [
-            'apikey=',
-            'instanceid=',
-            'region=',
-            'bucketname=',
-            'infile=',
-            'objectname=',
-            ])
-    except getopt.GetoptError:
-        print(help_message)
-        sys.exit(2)
-    for (opt, arg) in opts:
-        if opt == '-h':
-            print(help_message)
-            sys.exit()
-        elif opt in ('-k', '--apikey'):
-            apikey = arg
-        elif opt in ('-i', '--instanceid'):
-            instanceid = arg
-        elif opt in ('-r', '--region'):
-            region = arg
-        elif opt in ('-b', '--bucketname'):
-            bucketname = arg
-        elif opt in ('-f', '--infile'):
-            infile = arg
-        elif opt in ('-o', '--objectname'):
-            objectname = arg
-    cos = ibm_boto3.client(
-        service_name='s3',
-        ibm_api_key_id=apikey,
-        ibm_service_instance_id=instanceid,
-        ibm_auth_endpoint='https://iam.cloud.ibm.com/identity/token',
-        config=Config(signature_version='oauth'),
-        endpoint_url='https://s3.' + region
-            + '.cloud-object-storage.appdomain.cloud/',
+def main():
+    parser = argparse.ArgumentParser(
+        epilog="Note: Create a Service credential in the COS to get the apikey and the resource_instance_id")
+    parser.add_argument("-b", "--bucket",
+                        help="Bucket name",
+                        required=True)
+    parser.add_argument("-r", "--region",
+                        help="Bucket's region(e.g: us-south)",
+                        required=True)
+    parser.add_argument("-f", "--file",
+                        help="Input file to be uploaded",
+                        required=True)
+    parser.add_argument("-o", "--object",
+                        help="Target object name to be created in the bucket",
+                        required=True)
+    parser.add_argument("-a", "--accesskey",
+                        help="Storage Bucket's Access Key ID(not required if --aspera set")
+    parser.add_argument("-s", "--secret",
+                        help="Storage Bucket's Secret Access Key(not required if --aspera set)")
+    parser.add_argument("--aspera",
+                        help="Use aspera client to upload an image(available only in x86 linux environment)",
+                        action="store_true",
+                        default=False)
+    parser.add_argument("-k", "--apikey",
+                        help="IBM Cloud API Key(required if --aspera set)")
+    parser.add_argument("-i", "--instanceid",
+                        help="IBM COS resource instance ID(required if --aspera set)")
+
+    args = parser.parse_args()
+    if args.aspera:
+        print("Aspera option is set..")
```

Comment:
Possible to add a system arch check to ensure it only gets used for Intel ?

Reply:
fixed, ptal